Library/Homebrew/language/python.rb

```diff
@@ -285,10 +285,44 @@ def pip_install_and_link(targets)
 
         def do_install(targets)
           targets = Array(targets)
+          targets.map! { |target| substitute_poetry_wheel_if_needed(target) }
+
           @formula.system @venv_root/"bin/pip", "install",
                           "-v", "--no-deps", "--no-binary", ":all:",
                           "--ignore-installed", *targets
         end
+
+        # Generates a wheel from `poetry build` for Poetry-based targets, if
+        # needed, which can then be consumed by `pip install`.
+        #
+        # @param target [String, Pathname] the path to the target
+        # @return target [Pathname] the path to a replacement wheel target,
+        #   or the original input target if wheel generation is not necessary
+        def substitute_poetry_wheel_if_needed(target)
+          target_as_path = Pathname.new(target)
+          return target unless target_needs_poetry?(target_as_path)
+
+          old_files = Dir[target_as_path/"dist/*.whl"].to_set
+          @formula.system Formula["poetry"].opt_bin/"poetry", "build", "-vvv", "--format", "wheel"
```

Comment:
What happens if this fails?

Reply:
My intention is that the formula build will fail (i.e. a failure here is as severe as a failure in `pip install`ing a non-Poetry Python resource). My understanding is that `@formula.system` here will fail on non-zero exit codes, which should facilitate that.

Reply:
Gotcha 👍🏻